### DISCUSSION TITLE
Starting Capytaine with PyCharm

### DISCUSSION THREAD

**Discussion by Jake-Aus** [capytaine/capytaine#299] [Q&A]
Hi, 

I'm a bit new to python and totally new to capytaine. I'm sure this is a silly question, but I couldn't solve this yet. I appreciate any help :)

I have installed Anaconda and then installed capytaine in a separate environment from the Anaconda prompt. Next, I installed PyCharm, and opened a new project within the previously built environment using the python interpreter. I created a test file in that PyCharm project and started following the instruction given in capytaine documentation (Tutorial). But I get an error "_AttributeError: module 'capytaine' has no attribute 'mesh_sphere'_ ". What am I doing wrong?

**Comment by mancellin** [comment]
`mesh_sphere` was added in v1.5. Most likely an older version of Capytaine has been installed. You can check with
```python
from capytaine import __version__; print(__version__)
```
Possibly related to @accient's issue in https://github.com/capytaine/capytaine/discussions/276#discussioncomment-5210891

**Comment by Jake-Aus** [comment]
True, I have capytaine 1.4 :) I'm confused though, because I almost just installed Capytaine. Can I update this within Anaconda?

**Reply by mancellin** [reply]
That is weird. Creating a new conda environment with only Capytaine (as follow) should install the latest version.
```
conda create --name latest_capy -c conda-forge capytaine
conda activate latest_capy  # or set in PyCharm project configuration
```

If not, could you give me a bit more details about your setup?

**Reply by Jake-Aus** [reply]
Sure! I'm not sure what you want to see with the setup, so please let me know what should I provide here. In the meantime, I started installing a fresh copy too, following the installation instructions given by RubendeBruin in #111 ([link](https://github.com/capytaine/capytaine/issues/111#issuecomment-979329593))

**Reply by AlufaSam** [reply]
> That is weird. Creating a new conda environment with only Capytaine (as follow) should install the latest version.
> 
> ```
> conda create --name latest_capy -c conda-forge capytaine
> conda activate latest_capy  # or set in PyCharm project configuration
> ```
> 
> If not, could you give me a bit more details about your setup?

Hi, @mancellin. I also have a similar problem with the mesh "AttributeError: module 'capytaine' has no attribute 'mesh_sphere' ". I have tried the solution you proposed (and that proposed by https://github.com/capytaine/capytaine/issues/111#issuecomment-979329593), but I still get the version capytaine 1.4 installed all the time. I have tried to reinstall multiple times, and I have tried anaconda3 and both minicoda3. PS: I am new to the cmd style of programming.
![1 4](https://user-images.githubusercontent.com/105754974/225012226-c621116f-896d-4794-b164-b6e78201f816.png)

**Reply by AlufaSam** [reply]
I did check https://anaconda.org/conda-forge/capytaine/files and saw that I could download version 1.5. Can I point cmd or the power shell to the downloaded capytaine 1.5 version from a folder on my PC?

**Reply by mancellin** [reply]
What happens if you try to impose a version to conda?
```
conda create --name capy -c conda-forge capytaine==1.5
```

**Reply by mancellin** [reply]
> I did check https://anaconda.org/conda-forge/capytaine/files and saw that I could download version 1.5. Can I point cmd or the power shell to the downloaded capytaine 1.5 version from a folder on my PC?

I have no experience working with the `.conda` files that you can download directly from anaconda.org. Since running Capytaine also requires the installation of other packages such as Numpy and Scipy, it could be tedious to try to install everything manually. I would recommend sticking with `conda install` or `conda create` commands.

**Reply by AlufaSam** [reply]
> What happens if you try to impose a version to conda?
> 
> ```
> conda create --name capy -c conda-forge capytaine==1.5
> ```

Thank you for getting back to me so quickly. I tried your suggestions, and I, unfortunately, got some error while trying to check for the version of capytain installed. 
![error message](https://user-images.githubusercontent.com/105754974/225039725-c1ca15cf-590a-4336-a277-78a2014bcefa.png)

**Reply by AlufaSam** [reply]
Using the conda list, I can see that I have the capytaine version 1.5. But I get an error message when I try to access the version according to the tutorial in the documentation (https://ancell.in/capytaine/latest/user_manual/installation.html). Please do you know how I can access capytaine using the Jupyter QtConsole or Spyder? Thanks for your help so far.
![Picture3](https://user-images.githubusercontent.com/105754974/225042556-afd06a4f-4571-4f32-8cee-d185f7d0b87f.png)

**Reply by mancellin** [reply]
I guess the issue is that you are not working in the new environment that was just created by conda.

For Spyder, you can look at this section of their documentation: https://github.com/spyder-ide/spyder/wiki/Working-with-packages-and-environments-in-Spyder

For QtConsole, you can try to install and lauch it from the new environment with something like the following (untested):
```bash
conda activate capy  # or however the environment was named
conda install qtconsole  # will install in the current environment
jupyter qtconsole  # will launch in the current environment
```

**Reply by AlufaSam** [reply]
> ```shell
> conda install qtconsole
> ```

Thanks, now I have tried it and encountered the following error (attached image)..
![Picture4](https://user-images.githubusercontent.com/105754974/225051001-df1d2883-1d5e-4a4e-a820-ad510a03f4d8.png)

**Reply by mancellin** [reply]
And what if you require both Capytaine and QtConsole from the start when creating the environment?
```
conda create --name capy_and_qtconsole -c conda-forge capytaine==1.5 qtconsole
conda activate capy_and_qtconsole
jupyter qtconsole
```

**Reply by AlufaSam** [reply]
> And what if you require both Capytaine and QtConsole from the start when creating the environment?
> 
> ```
> conda create --name capy_and_qtconsole -c conda-forge capytaine==1.5 qtconsole
> conda activate capy_and_qtconsole
> jupyter qtconsole
> ```

Now it works. Thank you so much for your help.

**Comment by Jake-Aus** [comment]
Seems the problem is solved with the fresh copy as you suggested! Thank you!
